### PR TITLE
Commented Smart Rockets located in challenges

### PR DESCRIPTION
**Pull request by puruj**
Documented the java script files of the smart rockets challenge

### CODE REVIEW

File: challenges/CC_29_SmartRockets/dna.js [hunk 1]
```diff
@@ -4,31 +4,43 @@
 // Code for: https://youtu.be/bGz7mv2vD6g
 
 function DNA(genes) {
+  //recieves genes and create a dna object
```
Comment (shiffman) [inline challenges/CC_29_SmartRockets/dna.js]: Small thing but I prefer a comment style which starts with a capital letter and has a space after `//` like:

```javascript
// Receives genes and creates a dna object
```

File: challenges/CC_29_SmartRockets/dna.js [hunk 2]
```diff
@@ -4,31 +4,43 @@
...
     this.genes = [];
     for (var i = 0; i < lifespan; i++) {
+      //gives random vectors
       this.genes[i] = p5.Vector.random2D();
+      //sets maximum force of vector to be applied to a rocket
       this.genes[i].setMag(maxforce);
     }
   }
-
+  // decides childs fitness
```
Comment (shiffman) [inline challenges/CC_29_SmartRockets/dna.js]: This is not related to fitness but rather creates a new child based on two parent DNA objects.

File: challenges/CC_29_SmartRockets/dna.js [hunk 3]
```diff
@@ -4,31 +4,43 @@
...
+      //if i < mid new gene should come from other partners gene's 
+      else {
         newgenes[i] = partner.genes[i];
       }
     }
+    // gives DNA object an array
     return new DNA(newgenes);
   }
 
+  // allows child rockets to have path based on parent vectors
```
Comment (shiffman) [inline challenges/CC_29_SmartRockets/dna.js]: This randomly mutates vectors to add more variation

File: challenges/CC_29_SmartRockets/population.js [hunk 4]
```diff
@@ -4,52 +4,67 @@
...
     for (var i = 0; i < this.popsize; i++) {
+      //calculates fitness
       this.rockets[i].calcFitness();
+      //if current fitness is greater than max, then make max eqaul to current
       if (this.rockets[i].fitness > maxfit) {
         maxfit = this.rockets[i].fitness;
       }
     }
-
+    //divides current fitness by max and makes that the current fitness
```
Comment (shiffman) [inline challenges/CC_29_SmartRockets/population.js]: normalizes all fitness values

**Comment by puruj** [comment]
Okay thank you for the input I'll try fixing it up

**Comment by meiamsome** [comment]
I fixed this up and merged. Thanks, @puruj

**Pull request merged**